python-package/lightgbm/sklearn.py

```diff
@@ -811,7 +829,7 @@ def _get_meta_data(collection, name, i):
             num_boost_round=self.n_estimators,
             valid_sets=valid_sets,
             valid_names=eval_names,
-            feval=eval_metrics_callable,
+            feval=eval_metrics_callable,  # type: ignore[arg-type]
```

Comment:
This `# type: ignore` fixes this error:

```text
Argument "feval" to "train" has incompatible type "List[_EvalFunctionWrapper]"; expected "Union[Union[Callable[[ndarray[Any, Any], Dataset], Tuple[str, float, bool]], Callable[[ndarray[Any, Any], Dataset], List[Tuple[str, float, bool]]]], List[Union[Callable[[ndarray[Any, Any], Dataset], Tuple[str, float, bool]], Callable[[ndarray[Any, Any], Dataset], List[Tuple[str, float, bool]]]]], None]"  [arg-type]
sklearn.py:814: note: "List" is invariant -- see https://mypy.readthedocs.io/en/stable/common_issues.html#variance
sklearn.py:814: note: Consider using "Sequence" instead, which is covariant
```

I'm not sure why, but `mypy` isn't able to figure out that `_EvalFunctionWrapper` is actually a `Callable[[ndarray, Dataset], Tuple[str, float, bool]]`.

I think that's maybe related to some issues it has with comparing lists to union types including lists? e.g. https://github.com/python/mypy/issues/6463

This PR proposes just ignoring that error for now... the use of custom eval metric functions is well-covered by the unit tests in `test_sklearn.py` and `test_dask.py`.